src/Features/OnboardingTasks/Tasks/WooCommercePayments.php

```diff
@@ -107,8 +107,9 @@ public function can_view() {
 	 */
 	public static function is_requested() {
 		$profiler_data       = get_option( Onboarding::PROFILE_DATA_OPTION, array() );
+		$product_types       = isset( $profiler_data['product_types'] ) ? $profiler_data['product_types'] : array();
 		$business_extensions = isset( $profiler_data['business_extensions'] ) ? $profiler_data['business_extensions'] : array();
-		return in_array( 'woocommerce-payments', $business_extensions, true );
+		return in_array( 'woocommerce-payments', $business_extensions, true ) || in_array( 'subscriptions', $product_types, true );
```

Comment:
This is checking if the product type is `subscriptions` but it should also check if the store is based in the `US`.

`WCPay` will be automatically installed during the OBW for:
- US stores
- That selected `Subscriptions` in the `Product Types` step 
- It will happen after pressing `Continue` in the `Product Types` step.

Currently, it's still working because `self::is_supported` is [doing that check](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/OnboardingTasks/Tasks/WooCommercePayments.php#L99), but I think that we should check that here in order to make this method more reliable.

Reply:
Good catch @octaedro ! Fixed in https://github.com/woocommerce/woocommerce-admin/pull/8445/commits/96bfd31a773e199db073e40eed7ac2115ee34bc7.